### PR TITLE
Better error message when user is the wrong role

### PR DESCRIPTION
**Pull request by ewels**
- [ ] Tests passing
- [ ] Black formatting
- [ ] Migrations for any changes to the database schema
- [ ] Rebase/merge the `dev` branch
- [ ] Note in the CHANGELOG

### CODE REVIEW

File: dds_web/api/user.py [hunk 1]
```diff
@@ -271,8 +271,7 @@ def add_to_project(whom, project, role, send_email=True):
             return {
                 "status": ddserr.AccessDeniedError.code.value,
                 "message": (
-                    "User Role should be either 'Project Owner' or "
-                    "'Researcher' to be added to a project"
+                    f"This user is registered as a '{role}'. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```
Comment (talavis) [inline dds_web/api/user.py:274]: Do we want to leak the role of the user?

Comment (talavis) [reply dds_web/api/user.py:274]:
```suggestion
                    f"This user is registered as an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```

Comment (ewels) [reply dds_web/api/user.py:274]: How many roles are there? Is it not fairly obvious already?

Comment (talavis) [reply dds_web/api/user.py:274]: Maybe a non-issue. I have just become a bit paranoid.

Comment (ewels) [reply dds_web/api/user.py:274]: No it's a good point. Printing the role doesn't add much, I'll change it.

```suggestion
                    f"This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```

File: dds_web/api/user.py [hunk 2]
```diff
@@ -271,8 +271,7 @@ def add_to_project(whom, project, role, send_email=True):
             return {
                 "status": ddserr.AccessDeniedError.code.value,
                 "message": (
-                    "User Role should be either 'Project Owner' or "
-                    "'Researcher' to be added to a project"
+                    f"This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```
Comment (inaod568) [inline dds_web/api/user.py:274]:
```suggestion
                    "This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```

File: dds_web/api/user.py (hunk 1 above)
Comment (inaod568) [reply dds_web/api/user.py]:
```
root@6f52f567f7ab:/code# dds user add unituser2@mailtrap.io --role Researcher -p someunit00001
     ︵ 
 ︵ (  )   ︵ 
(  ) ) (  (  )   SciLifeLab Data Delivery System 
 ︶  (  ) ) (    http://dds_backend:5000/ 
      ︶ (  )    Version 0.0.11 
          ︶
Current user: unituser_1
ERROR    ⚠️ Could not add user: This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project. ⚠️    

                           
root@6f52f567f7ab:/code# dds user add unituser2@mailtrap.io --role "Unit Personnel" -p someunit00001
     ︵ 
 ︵ (  )   ︵ 
(  ) ) (  (  )   SciLifeLab Data Delivery System 
 ︶  (  ) ) (    http://dds_backend:5000/ 
      ︶ (  )    Version 0.0.11 
          ︶
Current user: unituser_1
ERROR    ⚠️ Could not add user: This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project. ⚠️   
```

So these two give the same response, but I think the `This user is already registered with an incompatible role` can cause a bit of confusion.

Comment (inaod568) [reply dds_web/api/user.py]: For
```
dds user add unituser2@mailtrap.io --role "Unit Personnel" -p someunit00001
```
it should just say something in the lines of "You cannot add a user with the role Unit Personnel to a project."

**Comment by inaod568** [comment]
This is solved in: https://github.com/ScilifelabDataCentre/dds_web/pull/1039

File: dds_web/api/user.py [hunk 3]
```diff
@@ -271,8 +271,7 @@ def add_to_project(whom, project, role, send_email=True):
             return {
                 "status": ddserr.AccessDeniedError.code.value,
                 "message": (
-                    "User Role should be either 'Project Owner' or "
-                    "'Researcher' to be added to a project"
+                    "This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```
[the remark this answers is not in the document]
Comment (ewels) [reply dds_web/api/user.py:274]: Sure 👍🏻

**Pull request closed without merging**